tests/integration/source_code/test_libraries.py

```diff
@@ -1,15 +1,39 @@
+"""
+These tests uses part of unit testing framework to mock the path lookup, and the only reason it"s integration tests
+because it uses the context and the time it takes to run the test.
+"""
+
 from pathlib import Path
 
+import pytest
+
 from tests.unit.conftest import MockPathLookup
 
 
-def test_loads_pip_library_from_notebook(simple_ctx):
-    # this tests uses part of unit testing framework to mock the path lookup,
-    # and the only reason it's integration tests because it uses the context
-    # and the time it takes to run the test.
+@pytest.mark.parametrize(
+    "notebook",
+    (
+        "pip_install_demo_wheel",
+        "pip_install_demo_wheel_and_pytest",
+        "pip_install_demo_wheel_many_flags",
+        "pip_install_demo_wheel_with_target_directory",
+    ),
+)
+def test_build_notebook_dependency_graphs_installs_wheel_with_pip_cell_in_notebook(simple_ctx, notebook):
+    ctx = simple_ctx.replace(path_lookup=MockPathLookup())
+
+    maybe = ctx.dependency_resolver.build_notebook_dependency_graph(Path(notebook))
+
+    assert not maybe.problems
+    assert maybe.graph.all_relative_names() == {f"{notebook}.py", "thingy/__init__.py"}
+
+
+@pytest.mark.xfail(reason="Spaces in path are not handled by subprocess when quoted")
```

Comment:
A pip install with spaces in a path does not work in subprocess despite quoting them. It works when running the same install in a terminal

Reply:
https://github.com/databrickslabs/ucx/issues/1859